test/converters/xcresult3/model3/conversion.go

```diff
@@ -150,3 +161,41 @@ func extractFailureMessage(testNode TestNode) (string, []string) {
 
 	return failureMessage, warnings
 }
+
+func extractRetries(testNode TestNode, fallbackName string) ([]TestCase, []string, error) {
+	var retries []TestCase
+	var warnings []string
+
+	for _, child := range testNode.Children {
+		if child.Type == TestNodeTypeRepetition {
+			// Use the original test node's identifier, instead of the repetition's identifier (1, 2, ...).
+			identifierSplit := strings.Split(testNode.Identifier, "/")
+			if len(identifierSplit) != 2 {
+				// If the identifier does not contain a class name, we use the fallback name.
+				return nil, nil, fmt.Errorf("expected identifier to contain a '/' but got: %s", testNode.Identifier)
+			}
+			className := identifierSplit[0]
+			if className == "" {
+				// In rare cases the identifier is an empty string, so we need to use the test suite name which is the
+				// same as the first part of the identifier in normal cases.
+				className = fallbackName
+			}
+
+			message, failureMessageWarnings := extractFailureMessage(child)
+			if len(failureMessageWarnings) > 0 {
+				warnings = append(warnings, failureMessageWarnings...)
+			}
+
+			testCase := TestCase{
+				Name:      testNode.Name,
+				ClassName: className,
+				Time:      extractDuration(child.Duration),
+				Result:    child.Result,
+				Message:   message,
+			}
+			retries = append(retries, testCase)
```

Comment:
This part is the same as the one at the end of the `extractTestCases` function. Maybe we could share this between the two.